src/webhook_payload_template.js

```diff
@@ -0,0 +1,164 @@
+ *    "dataToSend": {{data}}
+ * }`
+ *
+ * const data = {
+ *     status: 200,
+ *     body: 'hello world'
+ * }
+ *
+ * const payloadObject = WebhookPayloadTemplate.parse(payloadTemplate, null, { data })
+ * ```
+ *
+ * **Produces:**
+ * ```js
+ * {
+ *     id: "some-id",
+ *    createdAt: '2019-05-08T15:22:21.095Z',
+ *    dataToSend: {
+ *        status: 200,
+ *        body: 'hello world'
+ *    }
+ * }
+ * ```
+ */
+export default class WebhookPayloadTemplate {
+    constructor(payloadTemplate, allowedVariables = null, context = {}) {
+        this.template = payloadTemplate;
+        this.allowedVariables = allowedVariables;
+        this.context = context;
+
+        this.payload = this.template;
+        this.replacedVariables = [];
+    }
+
+    /**
+     * Parse existing webhook payload template string into an object, replacing
+     * template variables using the provided context.
+     *
+     * Parse also validates the template structure, so it can be used
+     * to check validity of the template JSON and usage of allowedVariables.
+     * @param {string} payloadTemplate
+     * @param {Set<string>|null} [allowedVariables]
+     * @param {Object} [context]
+     * @return {object}
+     */
+    static parse(payloadTemplate, allowedVariables, context) {
+        const type = typeof payloadTemplate;
+        if (type !== 'string') throw new Error(`Cannot parse a ${type} payload template.`);
+        const template = new WebhookPayloadTemplate(payloadTemplate, allowedVariables, context);
+        return template._parse(); // eslint-disable-line no-underscore-dangle
+    }
+
+    /**
+     * Stringify an object into a webhook payload template.
+     * Values created using `getTemplateVariable('foo.bar')`
+     * will be stringified to `{{foo.bar}}` template variable.
+     * @param {Object} objectTemplate
+     * @return {string}
+     */
+    static stringify(objectTemplate) {
+        const type = typeof objectTemplate;
+        if (!objectTemplate || type !== 'object') throw new Error(`Cannot stringify a ${type} payload template.`);
+        return jsonStringifyExtended(objectTemplate);
+    }
+
+    /**
+     * Produces an instance of a template variable that can be used
+     * in objects and will be stringified into `{{variable}}` syntax.
+     *
+     * **Example:**
+     * ```js
+     * const resourceVariable = WebhookPayloadTemplate.getVariable('resource');
+     * const objectTemplate = {
+     *     foo: 'foo',
+     *     bar: ['bar'],
+     *     res: resourceVariable,
+     * }
+     *
+     * const payloadTemplate = WebhookPayloadTemplate.stringify(objectTemplate);
+     * ```
+     *
+     * **Produces:**
+     * ```json
+     * {
+     *     "foo": "foo",
+     *     "bar": ["bar"],
+     *     "res": {{resource}},
+     * }
+     * ```
+     *
+     * @param {string} variable
+     * @return {JsonToken}
+     */
+    static getVariable(variable) {
+        return new JsonToken(variable);
```

Comment:
Is it really necessary to define such a simple function for this?

Reply:
It's an encapsulation thing. You don't want to do

```
const { jsonToken, jsonStringifyExtended } = require('apify-shared/utilities.client');
const WebhookPayloadTemplate = require('apify-shared/webhook_payload_template');

const var = new jsonToken('var');
```

when you can just think about having the one class and not even know about the `jsonToken` which is something else entirely.